sapmprotocol/parser.go

```diff
@@ -54,12 +54,20 @@ var (
 
 // ParseTraceV2Request processes an http request request into SAPM
 func ParseTraceV2Request(req *http.Request) (*splunksapm.PostSpansRequest, error) {
+	var sapm = &splunksapm.PostSpansRequest{}
+	if err := ParseSapmRequest(req, sapm); err != nil {
+		return nil, err
+	}
+	return sapm, nil
+}
+
+// ParseSapmRequest parses an http request request into an SAPM compatible proto definition.
+func ParseSapmRequest(req *http.Request, into proto.Unmarshaler) error {
```

Comment:
Do I understand it correct that there is no logic change, you just extracted ParseSapmRequest as a public function?

Reply:
Correct. One small change but not sure if it matters, we do allocate the object in advance so if there is an error we do an extra allocation. I don't think that should matter.

Reply:
I think it doesn't matter.